tests/rest/client/v2_alpha/test_sync.py

```diff
@@ -314,6 +315,144 @@ def test_sync_backwards_typing(self):
+    servlets = [
+        synapse.rest.admin.register_servlets,
+        login.register_servlets,
+        room.register_servlets,
+        sync.register_servlets,
+        knock.register_servlets,
+    ]
+
+    def prepare(self, reactor, clock, hs):
+        self.store = hs.get_datastore()
+        self.url = "/sync?since=%s"
+        self.next_batch = "s0"
+
+        # Register the first user (used to create the room to knock on).
+        self.user_id = self.register_user("kermit", "monkey")
+        self.tok = self.login("kermit", "monkey")
+
+        # Create the room we'll knock on.
+        self.room_id = self.helper.create_room_as(
+            self.user_id,
+            is_public=False,
+            room_version="xyz.amorgan.knock",
+            tok=self.tok,
+        )
+
+        # Register the second user (used to knock on the room).
+        self.knocker = self.register_user("knocker", "monkey")
+        self.knocker_tok = self.login("knocker", "monkey")
+
+        # Perform an initial sync for the knocking user
+        request, channel = self.make_request(
+            "GET", self.url % self.next_batch, access_token=self.tok,
+        )
+        self.assertEqual(channel.code, 200, channel.json_body)
+
+        # Store the next batch for the next request.
+        self.next_batch = channel.json_body["next_batch"]
+
+        # Set up some room state to test with.
+
+        # To set a canonical alias, we'll need to point an alias at the room first.
+        canonical_alias = "#fancy_alias:test"
+        self.get_success(
+            self.store.create_room_alias_association(
+                RoomAlias.from_string(canonical_alias), self.room_id, ["test"]
+            )
+        )
+
+        self.room_state = {
+            # We need to set the room's join rules to allow knocking
+            EventTypes.JoinRules: {
+                "content": {"join_rule": "xyz.amorgan.knock"},
+                "state_key": "",
+            },
+            # The rest are state events that are recommended to strip and send to clients
+            EventTypes.Name: {"content": {"name": "A cool room"}, "state_key": ""},
+            EventTypes.RoomAvatar: {
+                "content": {
+                    "info": {
+                        "h": 398,
+                        "mimetype": "image/jpeg",
+                        "size": 31037,
+                        "w": 394,
+                    },
+                    "url": "mxc://example.org/JWEIFJgwEIhweiWJE",
+                },
+                "state_key": "",
+            },
+            EventTypes.RoomEncryption: {
+                "content": {"algorithm": "m.megolm.v1.aes-sha2"},
+                "state_key": "",
+            },
+            EventTypes.CanonicalAlias: {
+                "content": {"alias": canonical_alias, "alt_aliases": []},
+                "state_key": "",
+            },
+        }
+
+        for event_type, event in self.room_state.items():
```

Comment:
Does the ordering of these matter at all? (dictionaries aren't ordered in older Python versions)

Reply:
Oh yes, the membership event should be last.